frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -795,18 +815,24 @@ pub mod pallet {
                     contract_id.clone(),
                     staker_reward,
                 ));
+
+                weight_info = T::WeightInfo::claim_staker_with_restake();
             }
 
-            T::Currency::resolve_creating(&staker, reward_imbalance);
+            // Withdraw reward funds from the dapps staking pot
+            let reward_imbalance = T::Currency::withdraw(
+                &Self::account_id(),
+                staker_reward,
+                WithdrawReasons::TRANSFER,
+                ExistenceRequirement::AllowDeath,
+            )?;
+
+            let reward_account = beneficiary.clone().unwrap_or(staker.clone());
+            T::Currency::resolve_creating(&reward_account, reward_imbalance);
             Self::update_staker_info(&staker, &contract_id, staker_info);
-            Self::deposit_event(Event::<T>::Reward(staker, contract_id, era, staker_reward));
+            Self::deposit_event(Event::<T>::Reward(reward_account, contract_id, era, staker_reward));
```

Comment:
Perhaps this is confusing a bit - `Reward` event would be read as `X staked on Y and received Z amount of reward for era N`.

Right now, beneficiary could be an account that's not even part of dapps-staking, so it'd be even more confusing.

Could you share your opinion on this?
Is there an alternative solution you could propose?

Reply:
Current Reward event reads as `someone staked on Y and its beneficiary is X, X received Z amount of reward for era N.` 
Adding an extract field called `staker` makes sense for easily index on staker for reward events.

For UX, most users don't need to bother this beneficiary settings, by default the `restake` should just works fine. User only need to care about beneficiary if they are making a staking pool for others, delegate and beneficiary will make it clear in this case.

Instead of adding an extra storage, make the reward destination to include beneficiary seems more straight forward, but require more thorough design and changes.

Reply:
Sure, I like the scenarios, even though this is just a dummy-pre-screening-task feature :)

Differentiating when stakers receive rewards and when it's just via a reward beneficiary is important if we want to be aware how much someone earned as a staker. With the modified event, we cannot tell whether the user received rewards as a beneficiary or as a nomination reward.

And love the proposal about reward destination :+1: :smile: , was hoping for it!

Reply:
Sure, I like the scenarios, even though this is just a dummy-pre-screening-task feature :)

Differentiating when stakers receive rewards and when it's just via a reward beneficiary is important if we want to be aware how much someone earned as a staker. With the modified event, we cannot tell whether the user received rewards as a beneficiary or as a nomination reward.

And love the proposal about reward destination :+1: :smile: , was hoping for it!